sketch/hash_base.hpp

```diff
@@ -47,24 +47,21 @@ class HashBase {
         switch (hash_algorithm) {
             case HashAlgorithm::uniform: {
                 T val;
-                // TODO multiple read Semaphor instead of critical
+                // TODO multiple read Semaphore instead of critical
 #pragma omp critical
                 {
-                    if (hashes[index].find(key) != hashes[index].end()) {
-                        val = hashes[index][key];
+                    auto [it, inserted] = hashes[index].insert({ key, -1 });
+                    if (not inserted) {
```

Comment:
I have nothing against using `not`, `or`, `and`, etc., but we should use them consistently. Since the rest of the code uses `!` rather than `not`, I think we should keep using it here, too.

Reply:
oh right; i've been writing them out since learning python, but I'll try not to do it here :)